### PR TITLE
[Main Course] - Argparse

### PR DESCRIPTION
**Pull request by jz263**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Python argparse 模块介绍
+
+## 1 前言
+
+Python 作为脚本语言，可以简单地在命令行中进行启动，但有的时候我们往往需要传入一些参数让同一套程序解决更多类型的任务。例如在 [12306 智能订票](https://bullettech.github.io/BulletTech/Main_Course/One_Day_Build/2021-11-10-12306ticket/)项目中，脚本根据命令行中传入参数的不同，有三种不同的运行模式，如:
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:5]: 可以换成微信文章

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+argparse 的使用主要分为三步：
+
+1. 实例化对象
+2. 使用 add_argument() 方法定义可以传递的参数
+3. 使用 parse_args() 方法将参数对象传入程序
+
+### 2.1 传递位置参数
+
+位置参数是在命令行中唤起脚本时，必须携带的参数。在 add_argument()方法中定义时，参数名称前缀没有"-"或"--"。一般用于传递执行程序时必不可少的参数。
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:25]: 好像最后一句话有点语病

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+if __name__ == "__main__":
+    # 创建解析器对象
+    parser = create_parser()
+    # 解析命令行参数
+    args = parser.parse_args()
+    # 调用函数
+    print("return value: ",compute_square(args.number))
+```
+
+我们设置改文件名为 `ap.py`。当我们在终端中启动该文件并传入一个整数时，程序将会打印该数的平方:
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:53]: 该

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+Area: 12
+```
+
+这样一来，我们不需要改动脚本内的代码，就可以简洁地在终端内实现重要参数的控制，提高代码的有效性。
+
+## 3. 小结
+
+除了常用的方法以外，argparse 模块还包含
+
+- add_argument_group():添加参数组
```
Comment (jhqian0909) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:141]: ：后面可以加个空格

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md (hunk 1 above)
Comment (jz263) [reply docs/Main_Course/Coding/2021-11-21-Python-argparse.md]: 已修改

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md (hunk 2 above)
Comment (jz263) [reply docs/Main_Course/Coding/2021-11-21-Python-argparse.md]: 已修改

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md (hunk 3 above)
Comment (jz263) [reply docs/Main_Course/Coding/2021-11-21-Python-argparse.md]: 已修改

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md (hunk 4 above)
Comment (jz263) [reply docs/Main_Course/Coding/2021-11-21-Python-argparse.md]: 已修改

**Review by jhqian0909 — APPROVED** [review]

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
+# Python argparse 模块介绍
+
```
Comment (Realvincentyuan) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:2]: 请补充作者信息

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+## 1 前言
+
+Python 作为脚本语言，可以简单地在命令行中进行启动，但有的时候我们往往需要传入一些参数让同一套程序解决更多类型的任务。例如在 [12306 智能订票](http://mp.weixin.qq.com/s?__biz=MzI4Mjk3NzgxOQ==&mid=2247484848&idx=1&sn=7e5b0b4e4740c42fa629a2c98e159839&chksm=eb90f6c4dce77fd266441c22a659668af98bc34e1bea9c264a296bc89a4f5c7da3570c6cb1e8#rd)项目中，脚本根据命令行中传入参数的不同，有三种不同的运行模式，如:
```
Comment (Realvincentyuan) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:5]: 这篇文章所有的英文左右都有空格，但之前没有，是特意更改的吗

File: docs/Main_Course/Coding/2021-11-21-Python-argparse.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    parser = create_parser()
+    # 解析命令行参数
+    args = parser.parse_args()
+    # 获取参数字典
+    print('args dict:', vars(args))
+    # 调用函数
+    print(compute_area(**vars(args)))
+```
+
+注意这里分别用了 add_argument()方法中一些常用参数来实现更规范和流畅的参数传:
```
Comment (Realvincentyuan) [inline docs/Main_Course/Coding/2021-11-21-Python-argparse.md:121]: 参数传递

**Review by Realvincentyuan — COMMENTED** [review]
谢谢Jeremy！

**Pull request merged**